processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -342,6 +346,12 @@ public void close()
         spatialWriter.close();
       }
 
+      if (dictionaryMergeBufferAllocations != null) {
```

Comment:
Made the dictionary merge iterator a `CloseableIterator` and added it to the try-with block in `writeIndexes()`